modules/orchestrator/src/main/java/org/eclipse/xpanse/orchestrator/OrchestratorService.java

```diff
@@ -78,278 +77,76 @@ public void onApplicationEvent(ApplicationEvent event) {
      */
-    public void updateManagedService(String managedServiceName, String oclLocation)
-            throws Exception {
-        Ocl ocl = oclLoader.getOcl(new URL(oclLocation));
-        updateManagedService(managedServiceName, ocl);
-    }
-
-    /**
-     * Update existing managed service with a new/updated OCL descriptor.
-     *
-     * @param managedServiceName the managed service to update, identified by the given name.
-     * @param ocl                the new/update OCL descriptor.
-     */
-    public void updateManagedService(String managedServiceName, Ocl ocl) {
-        MDC.put(TASK_TYPE, TaskType.UPDATE.toValue());
-
-        if (!managedServiceName.equals(ocl.getName())) {
-            log.warn("Can not update managed service {} with {}", managedServiceName,
-                    ocl.getName());
-        }
-
-        if (!this.databaseOrchestratorStorage.isExists(managedServiceName)) {
-            throw new EntityNotFoundException(
-                    "Managed service " + managedServiceName + " not found");
-        }
-
-        for (OrchestratorPlugin orchestratorPlugin : plugins) {
-            ServiceStatusEntity serviceStatusEntity =
-                    this.databaseOrchestratorStorage.getServiceDetailsByNameAndPlugin(
-                            managedServiceName, orchestratorPlugin);
-            putServiceInfoIntoLogMdc(serviceStatusEntity);
-            try {
-                serviceStatusEntity.setServiceState(ServiceState.UPDATING);
-                this.databaseOrchestratorStorage.store(serviceStatusEntity);
-                orchestratorPlugin.updateManagedService(managedServiceName, ocl);
-                serviceStatusEntity.setServiceState(ServiceState.UPDATED);
-                this.databaseOrchestratorStorage.store(serviceStatusEntity);
-            } catch (RuntimeException exception) {
-                updateFailedServiceStatusEntity(serviceStatusEntity, exception);
-                this.databaseOrchestratorStorage.store(serviceStatusEntity);
-                throw exception;
-            }
-        }
+    public void storeDeployService(DeployResult deployResult) {
+        DeployServiceEntity deployServiceEntity = new DeployServiceEntity();
+        deployServiceEntity.setVersion(deployResult.getTask().getCreateRequest().getVersion());
+        deployServiceEntity.setName(deployResult.getTask().getCreateRequest().getName());
+        deployServiceEntity.setId(deployResult.getTask().getCreateRequest().getId());
+        deployServiceEntity.setCsp(deployResult.getTask().getCreateRequest().getCsp());
+        deployServiceEntity.setFlavor(deployResult.getTask().getCreateRequest().getFlavor());
+        deployServiceEntity.setProperty(deployResult.getTask().getCreateRequest().getProperty());
+        deployServiceEntity.setOcl(deployResult.getTask().getOcl());
+        deployServiceEntity.setDeployResourceEntity(deployResult.getResources());
+        this.entityManager.persist(deployServiceEntity);
```

Comment:
we already have the "store" methods in the JPA interface class. We can use that instead of the entity manager directly.

Reply:
The "store" methods will not create the related tables automatically. I must store the different tables separately using the "store" method.

Reply:
oh ok. Ideally the DB tables should be created with different standalone scripts (which we don't have and also not needed at the moment since we are just using H2 DB)